src/main/java/net/imglib2/loops/IntervalChunks.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * #%L
+ * ImgLib2: a general-purpose, multidimensional image processing library.
+ * %%
+ * Copyright (C) 2009 - 2018 Tobias Pietzsch, Stephan Preibisch, Stephan Saalfeld,
+ * John Bogovic, Albert Cardona, Barry DeZonia, Christian Dietz, Jan Funke,
+ * Aivar Grislis, Jonathan Hale, Grant Harris, Stefan Helfrich, Mark Hiner,
+ * Martin Horn, Steffen Jaensch, Lee Kamentsky, Larry Lindsey, Melissa Linkert,
+ * Mark Longair, Brian Northan, Nick Perry, Curtis Rueden, Johannes Schindelin,
+ * Jean-Yves Tinevez and Michael Zinsmaier.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imglib2.loops;
+
+import net.imglib2.FinalInterval;
+import net.imglib2.Interval;
+import net.imglib2.util.IntervalIndexer;
+import net.imglib2.util.Intervals;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.IntStream;
+
+public class IntervalChunks
+{
+	/**
+	 * Returns a list of disjoint intervals, that covers the given interval.
+	 * <p>
+	 * Due to the divisibility of whole numbers, it not always easy to
+	 * return the correct number of intervals, in that case this method
+	 * returns a larger number of intervals.
+	 */
+	public static List< Interval > chunkInterval( Interval interval, int numChunks )
+	{
+		long[] chunkDimensions = suggestCellSize( Intervals.dimensionsAsLongArray( interval ), numChunks );
+		return generateGrid( interval, chunkDimensions );
+	}
+
+	/**
+	 * Suggest a cell size that can be used, to divide an interval of the
+	 * given dimensions, into the given number of chunks.
+	 */
+	static long[] suggestCellSize( long[] dimensions, long numberOfCells )
```

Comment:
I would suggest changing "Cell" to "Chunk" here in parameter, method, and variable names, to make naming consistent

Reply:
done